lib/vanagon/project.rb

```diff
@@ -270,14 +270,25 @@ def has_provides?
       !get_provides.empty?
     end
 
-    # Collects the preinstall packaging actions for the project and it's components
-    # for the specified packaging state
+    # Checks that the pkg_state is valid, return vanagon error if invalid
+    #
+    # @param pkg_state [String] package state input
+    # @param function [String] the function that called
+    def check_pkg_state(pkg_state, method_name)
+      if pkg_state.empty? || !(["install", "upgrade"].include? pkg_state)
+        raise Vanagon::Error, "#{pkg_state} should be a string containing one of 'install' or 'upgrade' in #{method_name}"
+      end
+    end
```

Comment:
Using the suggested guidance, this would look like this:

def check_pkg_state(pkg_state, method_name)
  allowed_values = %w[install upgrade]
  return pkg_state if allowed_values.include? pkg_state
  raise Vanagon::Error,
        "#{pkg_state} should be a string containing one of #{allowed_values.join", "} in ##{method_name}"
end
```

You can skip the `#empty?` check, since the `#include?` check will catch that particular use case:

```ruby
>> %w[a b c d e f g].include? ""
=> false
```

Reply:
Oh, I forgot. You can ditch `method_name` as an argument, and just use `__callee__` to ask Ruby for whatever method called `#check_pkg_state`.

Reply:
__callee__ seems to be returning 'check_pkg_state' but I've decided to just remove that bit instead